### ISSUE TITLE
Schedule View

### ISSUE THREAD

**Issue by Linktheoriginal** [codeforgreenville/trolley-tracker#130]
There's a few issues out there about this, with how confusing the schedule view is.  I'm playing with schedule views for the web app right now, and am looking for some feedback.

![schedule1](https://cloud.githubusercontent.com/assets/574842/12705176/1eb8e950-c83a-11e5-9a11-abcaf06670d6.png)
![schedule2](https://cloud.githubusercontent.com/assets/574842/12705177/20ffee02-c83a-11e5-99a8-9aa3bda61137.png)
![schedule3](https://cloud.githubusercontent.com/assets/574842/12705178/2223fd8c-c83a-11e5-9cac-a45e259b3c5f.png)

I'm not sure how these views will play with the trolley webpage, where it's being iframed.  not sure how to test how the new controls will look before pushing the change.

**Comment by jeremywight** [comment]
Based on your screenshots, that's much better @linktheoriginal if you want
I have yeahthattrolley.com and trackthetrolley.com and if you have a second
iframe I can grab to test we could test on trackthetrolley.com

**Comment by greenlinktransit** [comment]
Be aware that the Well is only serviced when there's an event.  The schedule in the screenshot above makes it seems like the Well is serviced every Friday from 6:00 - 11:00 PM and every Saturday from 10:00 AM - 11:00 PM.

Also, just a heads up that we added a trolley stop on Friday, which is at 106 Augusta Street.  This stop is called the "Trolley Park & Ride" and has shelters for people to wait.

**Comment by bikeoid** [comment]
Good point on the "every Friday" observation.   Perhaps adding the month and day to the day of week?

**Comment by Linktheoriginal** [comment]
@jeremywight I'll have to set up a beta url when I get time.

The schedule data is being pulled from the API - as far as I know, the schedule in the API is weekly right now?

**Comment by bikeoid** [comment]
Yes, the API only returns the schedule by day of week.   Assigning a date in the app will become much more complicated, although the server could add the next date that the weekday will run.

 Also, if we only look for the upcoming week, it can still be confusing: special route deviations will likely be entered only once per week so they show up for the weekend.   So on Sunday, it will be showing Thursday-Saturday as Thu-Sat of the following week, but it's likely still set for the current weekend.  So I'm still neutral on the value of adding the month/day.

**Comment by ajhodges** [comment]
We talked a little bit at the last meeting about the schedule showing dates
vs day of the week.. I personally think it makes more sense from a user's
perspective to show them the schedule for the upcoming week but understand
that complicates things in the back-end. The scenario you mentioned
definitely happens with the current android app... but I'm of the opinion
that showing dates is a better user experience (even if they are 'faked'
like I'm currently doing them and sometimes inaccurate). Not sure what the
best solution actually is but I think for now just showing the dates for
the upcoming week is Good Enough™

**Comment by bikeoid** [comment]
I like the dates also.   This isn't hard to add at the server - since it's already working in the Android App, are there any requests to add the date to the schedule API?

**Comment by Linktheoriginal** [comment]
Jeremy, I just now set up the test url for the beta web tracker at http://linktheorigi.webfactional.com/trolley-test/

Can you review it on the other url and make sure that the new controls won't cause issues?

**Comment by jeremywight** [comment]
John: It looks good on my screen, though deepening on the size of the screen it could be covered by the phone image. If it were positioned bottom right instead of bottom left it wouldn’t get covered. Could you make that change @linktheoriginal

Live here: http://trackthetrolley.com/greenville-sc-trolley/

**Comment by jeremywight** [comment]
Another thing I noticed is that once you are in the schedule mode there is no way to exit back to the active trolley route, without refreshing the page. Probably worth creating an exit button or similar.

**Comment by jeremywight** [comment]
Just noticed the back arrow :)

**Comment by Linktheoriginal** [comment]
I put the schedule in the bottom left since I figured the righty majority wouldn't want it covered by their hand.  I could move the back arrow up with the schedule/info buttons, though I may need to style it more boldly.  I've had some issues with the framework I'm using to put info anywhere other than the corners.  Let me tinker with it and get back to you.

**Comment by nullvariable** [comment]
FYI the http://linktheorigi.webfactional.com/trolley/ link is getting a Gateway timeout error

**Comment by Linktheoriginal** [comment]
Looking into it.

**Comment by Linktheoriginal** [comment]
Locally run version is fine, so I believe it's a issue with my host.  Submitting a support ticket with them.

**Comment by Linktheoriginal** [comment]
@bikeoid While I wait on the support ticket, is there any way you can confirm with Walling Info Systems that they're not blocking anything from my host on their end?

**Comment by bikeoid** [comment]
Checking...

**Comment by Linktheoriginal** [comment]
My hosting support reports that they're getting the 504 from the API from their hosting servers.  They're opening up a ticket with their upstream provider, but it may be an issue on the API's end.

**Comment by jeremywight** [comment]
Great teamwork on this y'all!  BTW - Both iOS and web app seem to be
working for me.

**Comment by Linktheoriginal** [comment]
From my host:
"It appears there is a peering error between US SoftLayer servers to Canada OVH servers or at least several ones. I am waiting on confirmation from Softlayer."

As of 2:30 it looks to be back up.
